src/xdgiconloader/xdgiconloader.cpp

```diff
@@ -766,18 +770,45 @@ QPixmap PixmapEntry::pixmap(const QSize &size, QIcon::Mode mode, QIcon::State st
     if (basePixmap.isNull())
         basePixmap.load(filename);
 
+    // see QPixmapIconEngine::adjustSize
     QSize actualSize = basePixmap.size();
     // If the size of the best match we have (basePixmap) is larger than the
     // requested size, we downscale it to match.
     if (!actualSize.isNull() && (actualSize.width() > size.width() || actualSize.height() > size.height()))
         actualSize.scale(size, Qt::KeepAspectRatio);
 
+#if (QT_VERSION >= QT_VERSION_CHECK(6,8,0))
+    // see QIconPrivate::pixmapDevicePixelRatio
+    qreal calculatedDpr;
+    QSize targetSize = size * scale;
+    if ((actualSize.width() == targetSize.width() && actualSize.height() <= targetSize.height()) ||
+        (actualSize.width() <= targetSize.width() && actualSize.height() == targetSize.height()))
+    {
+        // Correctly scaled for dpr, just having different aspect ratio
+        calculatedDpr = scale;
+    }
+    else
+    {
+        qreal ratio = 0.5 * (qreal(actualSize.width()) / qreal(targetSize.width()) +
+                             qreal(actualSize.height() / qreal(targetSize.height())));
+        calculatedDpr = qMax(qreal(1.0), scale * ratio);
+    }
+
+    QString key = QLatin1String("$qt_theme_")
+                  % HexString<quint64>(basePixmap.cacheKey())
+                  % HexString<quint8>(mode)
+                  % HexString<quint64>(QGuiApplication::palette().cacheKey())
+                  % HexString<uint>(actualSize.width())
+                  % HexString<uint>(actualSize.height())
+                  % HexString<quint16>(qRound(calculatedDpr * 1000));
+#else
```

Comment:
Why the change in types of the key and/or why only applicable for Qt >= 6.8?
For the key assembly I would suggest to not repeat everything, but just add conditionally the new key, e.g.:
```cpp
    QString key = QLatin1String("lxqt_")
                  % filename
                  % HexString<quint8>(mode)
                  % HexString<int>(state)
                  % HexString<uint>(size.width())
                  % HexString<uint>(size.height())
#if (QT_VERSION >= QT_VERSION_CHECK(6,8,0))
                  % HexString<quint16>(qRound(scale * 1000))
#endif
                  ;
```
(the change would be better visible, IMO; applicable also in next changes)

Reply:
Good point!

Reply:
It's very simple: I didn't want to change *anything* for Qt < 6.8 here. I really prefer it this way because it reflects the changes.

Reply:
> It's very simple: I didn't want to change anything for Qt < 6.8 here.

Why did you change the types? There is nothing about it in commit message.